pkg/controller/podautoscaler/horizontal.go

```diff
@@ -50,8 +51,8 @@ const (
 )
 
 type HorizontalController struct {
-	scaleNamespacer unversionedextensions.ScalesGetter
-	hpaNamespacer   unversionedextensions.HorizontalPodAutoscalersGetter
+	scaleClient   *scaleclient.Client
```

Comment:
It could be very reasonable to implement the interface using a dynamic client, but the controller should define what it needs from its client in terms of an interface